File: June-2023/C++/SummaryRanges.cxx
// 228. Summary Ranges
// 2 pointer approach
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> result;
        if (nums.empty()) return result;
        if (nums.size() == 1) {
            result.push_back(to_string(nums[0]));
            return result;
        }
        int i = 0, j = 0;
        while (i < nums.size() && j < nums.size()) {
            if (j < nums.size() - 1 && nums[j] + 1 == nums[j + 1]) {
                j++;
            }
            else {
                string s;
                if (i == j) {
                    s = to_string(nums[i]);
                }
                else {
                    s = to_string(nums[i]) + "->" + to_string(nums[j]);
                }
                result.push_back(s);
                i = j + 1;
                j++;
            }
        }
        return result;
    }
};

File: June-2023/Java/SummaryRanges.java
// 228. Summary Ranges
// 2 pointer approach 
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> l = new LinkedList<>();
        if(nums.length==0) return new LinkedList<>();
        if(nums.length==1){
            l.add(""+nums[0]);
            return l;
        }
        int i=0, j=0;
        while(i<nums.length && j<nums.length){
            if(j<nums.length-1 && nums[j]+1==nums[j+1]){
                j++;
            }
            else{
                String s;
                if(i==j){
                  s = nums[i]+"";
                }
                else{
                s = nums[i]+"->"+nums[j];
                }
                l.add(s);
               
                i=j+1;
                j++;
            }
          
        }
          return l;
      }
   }
                
            
            
           
        


    
